adapters/appnexus/appnexus.go

```diff
@@ -301,21 +324,64 @@ func (a *AppNexusAdapter) MakeRequests(request *openrtb.BidRequest) ([]*adapters
 		return nil, errs
 	}
 
-	reqJSON, err := json.Marshal(request)
-	if err != nil {
-		errs = append(errs, err)
-		return nil, errs
+	// Add Appnexus request level extension
+	if len(request.Ext) > 0 {
+		var reqExt appnexusReqExt
+		if err := json.Unmarshal(request.Ext, &reqExt); err == nil {
+			includeBrandCategory := reqExt.Prebid.Targeting != nil && reqExt.Prebid.Targeting.IncludeBrandCategory.PrimaryAdServer != 0
+			if includeBrandCategory {
+				if reqExt.Appnexus == nil {
+					reqExt.Appnexus = &appnexusReqExtAppnexus{}
+				}
+				reqExt.Appnexus.BrandCategoryUniqueness = &includeBrandCategory
+				reqExt.Appnexus.IncludeBrandCategory = &includeBrandCategory
+				request.Ext, err = json.Marshal(reqExt)
+				if err != nil {
+					errs = append(errs, err)
+					return nil, errs
+				}
+			}
+		} else {
+			errs = append(errs, err)
+			return nil, errs
+		}
 	}
 
+	imps := request.Imp
+	initialCapacity := (len(imps) + maxImpsPerReq - 1) / maxImpsPerReq
```

Comment:
`maxImpsPerReq - 1` is so we round up instead of down?

Reply:
@hhhjort this is initial capacity for future array of requests, small memory optimization.
Let's say there are 35 impressions and limit impressions per request equals to 10. In this case we need to create 4 requests with 10, 10, 10 and 5 impressions. 
With this formula initial capacity=(35+10-1)/10 = 4
Please let us know if you see a better way to do this.

Reply:
It is fine for me, I made the comment mostly because it took me several minutes to figure out the purpose of adding that value in there, and thought the comment might save other reviewers a little time. And on the off chance there was something else going on that I wasn't seeing.

Reply:
Comment added